frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,24 @@
+import { Navigate } from 'react-router-dom';
+
+import { AppRoute } from '~/libs/enums/app-route.enum';
+import { useAppSelector } from '~/libs/hooks/hooks.js';
+
+type Properties = {
+  component: React.FC
+};
+
+const ProtectedRoute: React.FC<Properties> = ({ component: Component, ...rest }) => {
```

Comment:
Should not take `component` prop - should wrap routes from `children`
to use it like this:
```
<ProtectedRoute>
  <Page />
</ProtectedRoute>
```

```suggestion
const ProtectedRoute: React.FC<Properties> = ({ children }) => {
```